server/src/main/java/io/crate/execution/engine/indexing/BatchIteratorBackpressureExecutor.java

```diff
@@ -110,7 +110,11 @@ public BatchIteratorBackpressureExecutor(UUID jobId,
 
     private void continueConsumptionOrFinish(@Nullable R result, Throwable failure) {
         if (result != null) {
-            resultRef.accumulateAndGet(result, combiner);
+            try {
+                resultRef.accumulateAndGet(result, combiner);
+            } catch (RuntimeException e) {
+                consumptionFinished = true;
```

Comment:
Wouldn't it be enough to look for `failure` and abort the loop/processing if it is present and `failFast` is set. Without further extending the UpsertResultCollectors ?

Reply:
my understanding is that `failure` is only meant for handling exceptions thrown from `batchIterator.loadNextBatch`. I instead need access to `errorCount` from `UpsertResults`. That was the reason why I had to extend `UpsertResultCollectors`.